src/connection_string.ts

```diff
@@ -57,11 +57,11 @@ function matchesParentDomain(srvAddress: string, parentDomain: string): boolean
  */
 export function resolveSRVRecord(options: MongoOptions, callback: Callback<HostAddress[]>): void {
   if (typeof options.srvHost !== 'string') {
-    return callback(new MongoParseError('Cannot resolve empty srv string'));
+    return callback(new MongoAPIError('Option "srvHost" must not be empty'));
   }
 
   if (options.srvHost.split('.').length < 3) {
-    return callback(new MongoParseError('URI does not have hostname, domain name and tld'));
+    return callback(new MongoURIError('URI must include hostname, domain name, and tld'));
```

Comment:
Replace this with MongoAPIError and mark for replacement with MongoConnectionStringError in NODE-3484.